test/AngleTest.cpp

```diff
@@ -0,0 +1,222 @@
+#include <gtest/gtest.h>
+#include "roboteam_utils/Angle.h"
+
+namespace rtt {
+double EPSILON = 1e-6;          // Difference that is used to check whether the actual and expected value are similar.
+double SUPREME_EPSILON = 1e-9;  // Small value that is used to create test cases
+
+TEST(AngleTests, rescale) {
+    /* Apply equivalence partitioning as testing method (https://en.wikipedia.org/wiki/Equivalence_partitioning) with the following partitions:
+     * angle < -PI, -PI <= angle < 0, angle = 0, 0 < angle < PI, angle >= PI
+     * and extend these partitions with boundary value analysis. We do not test angle = -FLT_MAX or angle = FLT_MAX, since we expect things to go wrong for these test cases, but
+     * we test other big positive and negative value instead. */
+
+    // Partition tests for: angle < -PI
+    double testAngle = -372349.9654;  // A random large negative number
+    double expectedRescale = -2.120912;
+    double actualRescale = Angle(testAngle).getAngle();
+    ASSERT_TRUE(fabs(actualRescale - expectedRescale) < EPSILON);
```

Comment:
Use ASSERT_NEAR(actualRescale,ExpectedRescale,EPSILON), it works better for some edge cases (floating point numbers are difficult and google tests helps you here)

Reply:
You don't need to change this though, just remember in the future. I'm convinced the code works fine.